File: app/src/main/java/com/app/l_pesa/common/CustomTypefaceSpan.java
package com.app.l_pesa.common;

/**
 * Created by Intellij Amiya on 3/2/19.
 * Who Am I- https://stackoverflow.com/users/3395198/
 * A good programmer is someone who looks both ways before crossing a One-way street.
 * Kindly follow https://source.android.com/setup/code-style
 */

import android.graphics.Paint;
import android.graphics.Typeface;
import android.text.TextPaint;
import android.text.style.TypefaceSpan;

public class CustomTypefaceSpan extends TypefaceSpan {

    private final Typeface newType;

    public CustomTypefaceSpan(String family, Typeface type) {
        super(family);
        newType = type;
    }

    @Override
    public void updateDrawState(TextPaint ds) {
        applyCustomTypeFace(ds, newType);
    }

    @Override
    public void updateMeasureState(TextPaint paint) {
        applyCustomTypeFace(paint, newType);
    }

    private static void applyCustomTypeFace(Paint paint, Typeface tf) {
        int oldStyle;
        Typeface old = paint.getTypeface();
        if (old == null) {
            oldStyle = 0;
        } else {
            oldStyle = old.getStyle();
        }

        int fake = oldStyle & ~tf.getStyle();
        if ((fake & Typeface.BOLD) != 0) {
            paint.setFakeBoldText(true);
        }

        if ((fake & Typeface.ITALIC) != 0) {
            paint.setTextSkewX(-0.25f);
        }

        paint.setTypeface(tf);
    }
}

File: app/src/main/java/com/app/l_pesa/common/CustomTypeFaceSpan.kt
package com.app.l_pesa.common

import android.graphics.Paint
import android.graphics.Typeface
import android.text.TextPaint
import android.text.style.TypefaceSpan

import androidx.annotation.ColorInt

class CustomTypeFaceSpan(family: String, private val newType: Typeface, @param:ColorInt @get:ColorInt
val foregroundColor: Int) : TypefaceSpan(family) {

    override fun updateDrawState(ds: TextPaint) {
        ds.color = foregroundColor
        applyCustomTypeFace(ds, newType)
        ds.isUnderlineText = false
    }

    override fun updateMeasureState(paint: TextPaint) {
        applyCustomTypeFace(paint, newType)
    }

    override fun getSpanTypeId(): Int {
        return super.getSpanTypeId()
    }


    private fun applyCustomTypeFace(paint: Paint, tf: Typeface) {
        val oldStyle: Int
        val old = paint.typeface
        if (old == null) {
            oldStyle = 0
        } else {
            oldStyle = old.style
        }

        //  int fake = oldStyle & ~tf.getStyle();
        /*if ((fake & Typeface.BOLD) != 0) {
            paint.setFakeBoldText(true);
        }*/

        /*if ((fake & Typeface.ITALIC) != 0) {
            paint.setTextSkewX(-0.25f);
        }
*/

        paint.typeface = tf
    }
}